multiuser/permission/che-multiuser-permission-workspace/src/main/java/org/eclipse/che/multiuser/permission/workspace/server/spi/jpa/MultiuserJpaWorkspaceDao.java

```diff
@@ -311,8 +315,35 @@ public void onCascadeEvent(BeforeAccountRemovedEvent event) throws Exception {
               (maxItems, skipCount) ->
                   workspaceManager.getByNamespace(
                       event.getAccount().getName(), false, maxItems, skipCount))) {
+        // Will try to stop running/starting workspaces.
+        // At the moment is impossible to force stop for  STOPPING workspace
+        if (RUNNING.equals(workspace.getStatus()) || STARTING.equals(workspace.getStatus())) {
+          String owner = workspace.getRuntime().getOwner();
+          if (owner.equals(EnvironmentContext.getCurrent().getSubject().getUserId())) {
+            workspaceManager.stopWorkspace(workspace.getId(), Collections.emptyMap());
```

Comment:
I don't understand how it's going to work since it's an asynchronous method and there is no guarantee that workspace is stopped at the time when removeWorkspace is called...
Maybe instead it makes sense to introduce async method to remove user? It'll mark user as inactive(not able to create/start more workspaces), stop all workspaces, and remove user along with all related objects once workspaces are stopped? It seems to be more complex than the proposed solution but I just don't understand how this PR is going to work. I think in most cases users will still get error because of try to remove non-stopped workspace.

Reply:
Looks like you are totally right, how i I miss that :(.  But i don't understand why it works for me on test cluster. 
As have proposal to add method to the WorkspaceManager like `stopAndRemove()`, it will remove workspace after stop. What do you think? @skabashnyuk @sleshchenko

Reply:
Maybe that's OK to stop every workspace asynchronous and throw an exception like `User had running workspaces which now are marked as stopping. You'll be unblocked to remove user when all workspaces are stopped. Please try again`
Also throw similar error when user has STOPPING workspaces.

Reply:
> As have proposal to add method to the WorkspaceManager like stopAndRemove(), it will remove workspace after stop. 

I don't think that's good idea because remove user is a synchronous request that typically will be timed out by clients in pretty short time while stop workspace is potentially can occupy like up to 5 minutes.